### PR TITLE
Product list implementation

### PR DESCRIPTION
**Pull request by DarinaBlazevica**
Product List Implementation: 

-Product list added for Bookstore
-Text symbol check added(more than 50 - dots are displayed)
-Move data to JSON file (title, etc.)
-Lazy loading for components(when click on book book-details view appears)
-For data fetching I used the same approach as in tutorial, but in case specific server is needed I can change that (JSON server)

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -6,4 +6,5 @@ import { Component } from '@angular/core';
   styleUrls: ['./app.component.css']
 })
 export class AppComponent {
+  title ='bookstore-angular'
```
Comment (Jalal999) [inline src/app/app.component.ts]: add access modifier

File: src/app/components/book-details/book-details.component.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  templateUrl: './book-details.component.html',
+  styleUrls: ['./book-details.component.css']
+})
+export class BookDetailsComponent implements OnInit {
+  
+  @Input() book?: Book;
+    constructor() { }
+  
+
+  ngOnInit(): void {
```
Comment (Jalal999) [inline src/app/components/book-details/book-details.component.ts:15]: if empty, can be removed

File: src/app/services/in-memory-data.service.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+
+export class InMemoryDataService implements InMemoryDbService{
+  createDb() {
+    const books = data.books;
+    return {books};
+  }
+  constructor() { }
```
Comment (Jalal999) [inline src/app/services/in-memory-data.service.ts]: if empty, can be removed

File: src/styles.css [hunk 4]
```diff
@@ -1,9 +1,4 @@
...
-html {
-    margin: 0;
-    padding: 0;
-}
-
-p {
-    padding: 0 10px 0 10px;
+body {
+    margin:0px;
+    padding:0px;
```
Comment (EgijaLocmele) [inline src/styles.css]: add spaces

File: src/app/components/main-page/main-page.module.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { MainPageRoutingModule } from './main-page-routing.module';
+import { MainPageComponent } from './main-page.component';
+
+
```
Comment (EgijaLocmele) [inline src/app/components/main-page/main-page.module.ts]: delete empty rows

File: src/app/components/book-details/book-details.component.ts (hunk 2 above)
Comment (DarinaBlazevica) [reply src/app/components/book-details/book-details.component.ts:15]: This is component for another part implementation and can be filled after in another PR(for book details)

File: src/app/app.component.ts (hunk 1 above)
Comment (DarinaBlazevica) [reply src/app/app.component.ts]: Added

File: src/app/services/in-memory-data.service.ts (hunk 3 above)
Comment (DarinaBlazevica) [reply src/app/services/in-memory-data.service.ts]: Deleted

File: src/styles.css (hunk 4 above)
Comment (DarinaBlazevica) [reply src/styles.css]: added

File: src/app/components/main-page/main-page.module.ts (hunk 5 above)
Comment (DarinaBlazevica) [reply src/app/components/main-page/main-page.module.ts]: Deleted

**Review by Jalal999 — APPROVED** [review]

**Review by EgijaLocmele — APPROVED** [review]

**Pull request merged**